SPECS/cert-manager/nested-vendoring.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# takes one argument: the directory to recursively vendor
+function vendor_all() {
+  # Check if the argument is a valid directory
+  if [ -d "$1" ]; then
+    # Save the current directory
+    local old_dir=$(pwd)
+    # Change to the given directory
+    cd "$1"
+    # Loop through all the go.mod files in the current directory and its subdirectories
+    for f in $(find . -name "go.mod"); do
+      # Push the directory containing the go.mod file to the stack
+      pushd $(dirname $f) > /dev/null
+      # Run go mod vendor
+      go mod vendor
```

Comment:
afaik `go mod vendor` also takes this argument 

```
The -o flag causes vendor to create the vendor directory at the given 
path instead of "vendor". The go command can only use a vendor directory 
named "vendor" within the module root directory, so this flag is 
primarily useful for other tools.
```

this could help merge the 2 separate functions into one, since we already have the vendor directory beforehand

Reply:
Nice catch, I'll refactor.

Reply:
Comments addressed, see latest commit.